Source/JavaScriptCore/inspector/remote/cocoa/RemoteConnectionToTargetCocoa.mm

```diff
@@ -238,7 +244,12 @@ static void RemoteTargetHandleRunSourceWithInfo(void* info)
 
 void RemoteConnectionToTarget::setupRunLoop()
 {
-    CFRunLoopRef targetRunLoop = m_target->targetRunLoop();
+    RefPtr<RemoteControllableTarget> target = nullptr;
+    {
+        Locker locker { m_targetMutex };
+        target = m_target.get();
+    }
```

Comment:
Simpler, but this may run afoul of the `WTF_GUARDED_BY_LOCK(m_targetMutex)` attribute without using `m_targetMutex`:

```suggestion
    auto target = m_target.get();
```